### PR TITLE
Detect prefilter events 6728 v1

### PR DESCRIPTION
**Pull request by catenacyber**
## Ticket

Redmine ticket: https://redmine.openinfosecfoundation.org/issues/6728

### CODE REVIEW

File: tests/decode-too-small/test.rules [hunk 1]
```diff
@@ -1,3 +1,4 @@
 alert tcp any any -> any any (msg:"TCP packet too small"; decode-event:tcp.pkt_too_small; sid:1;)
+alert tcp any any -> any any (msg:"TCP packet too small"; decode-event:tcp.pkt_too_small; prefilter; requires: version >= 8; sid:1; rev: 2;)
```
Comment (victorjulien) [inline tests/decode-too-small/test.rules:2]: sid 2 is already in this file

Comment (victorjulien) [reply tests/decode-too-small/test.rules:2]: ah no it's sid 1 rev 2

Comment (victorjulien) [reply tests/decode-too-small/test.rules:2]: any reason to use this pattern? Might as well have 2 rules side by side, to test both prefilter and non-prefilter path

Comment (catenacyber) [reply tests/decode-too-small/test.rules:2]: > any reason to use this pattern? 

This way, it works on both suricata 7 and suricata 8

If you have another rule side by side, you need the whole test to have min-version 8 as suricata 7 does not support prefilter on decode-event 

And this is also a good example/showcase of the `requires` keyword : you load the better rule if suricata version allows it

**Comment by catenacyber** [comment]
Continued in https://github.com/OISF/suricata-verify/pull/2001

**Pull request closed without merging**